src/brush.rs

```diff
@@ -29,18 +29,34 @@ impl From<Gradient> for Brush {
     }
 }
 
+impl From<Image> for Brush {
+    fn from(value: Image) -> Self {
+        Self::Image(value)
+    }
+}
+
 impl Default for Brush {
     fn default() -> Self {
         Self::Solid(Color::default())
     }
 }
 
+impl Brush {
+    /// Returns the brush with the alpha component multiplied by the specified
+    /// factor.
+    #[must_use]
+    pub fn with_alpha_factor(&self, alpha: f32) -> Self {
```

Comment:
I think this one should take `self` by value and perform the modifications in place. This lets the user decide if they want to clone the full brush.